src/JJMasterData.Commons/Data/DataAccess.cs

```diff
@@ -97,6 +99,7 @@ public DbProviderFactory Factory
     /// <summary>
     /// By default DataAccess recover connection string from appsettings.json with name ConnectionString
     /// </summary>
+    [Obsolete("Will be removed at Sun version, this constructor uses a static service locator and can cause runtime errors.")]
```

Comment:
We can't force it that way, nothing prevents using it without dependency injection

Reply:
I strongly disagree, this constructor is impossible to exist at .NET Core without the static service locator (`JJService`) and causes a lot of headache when people forget to call `UseJJMasterData` at `Program.cs` .

If you don't want to use dependency injection, this constructor is perfectly fine :
```cs
public DataAccess(string connectionString, string connectionProvider)
{
///....
}
```

If you really want that empty constructor, we can write a #ifdef to only work at .NET Framework, but in .NET Core it causes lots of problems.